src/core/builders/create-form-validator.ts

```diff
@@ -273,6 +275,20 @@ const getDependents = <Err>(
     }
   });
 
+const getParentsChain = <Err>(
+  desc: FieldDescriptor<any, Err>,
+  chain: FieldDescriptor<any, Err>[] = []
+): FieldDescriptor<any, Err>[] => {
```

Comment:
sorry for being picky, but I think version withhout mutable array passed between recursive calls is more elegant:


```ts
const getParentsChain = <Err>(
  desc: FieldDescriptor<any, Err>
): FieldDescriptor<any, Err>[] => {
  const parent = impl(desc).__parent;

  if (!parent) {
    return [];
  }

  const parentOpaque = opaque<unknown, Err>(parent);
  return [parentOpaque, ...getParentsChain(parentOpaque)];
};
```

Reply:
sure it is, mutability though is cheaper performance-wise (hopefully 😅 )
but since it depends on form's depth (which shouldn't be an issue, we can switch to non-mutable

Reply:
makes sense, we can stay with mutable; just wanted to throw my 5 cents in